engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/typeinformation/FlinkTypeInfoRegistrar.scala

```diff
@@ -25,7 +25,7 @@ object FlinkTypeInfoRegistrar {
     RegistrationEntry(classOf[LocalDateTime], classOf[LocalDateTimeTypeInfoFactory]),
   )
 
-  def ensureTypeInfosAreRegistered(): Unit = {
+  def ensureTypeInfosAreRegistered(): Unit = synchronized {
```

Comment:
I would add synchronized in the `register(entry)` method instead. In this method it is more visible that we are getting something from static, doing a check, and after that adding. Plus I would add a notice, that TypeExtractor has no method to register type info factory in the thread safety way

Reply:
Nice, thanks